naga/src/back/msl/writer.rs

```diff
@@ -1741,7 +1768,39 @@ impl<W: Write> Writer<W> {
                 {
                     write!(self.out, " ? ")?;
                     self.put_access_chain(expr_handle, policy, context)?;
-                    write!(self.out, " : DefaultConstructible()")?;
+                    write!(self.out, " : ")?;
+
+                    if context.resolve_type(base).pointer_space().is_some() {
+                        // We can't just use `DefaultConstructible` if this is a pointer.
+                        // Instead, we create a dummy local variable to serve as pointer
+                        // target if the access is out of bounds.
+                        let result_ty = context.info[expr_handle]
+                            .ty
+                            .inner_with(&context.module.types)
+                            .pointer_base_type();
+                        let result_ty_handle = match result_ty {
+                            Some(TypeResolution::Handle(handle)) => handle,
+                            Some(TypeResolution::Value(_)) => {
+                                // I don't have a succinct argument why this is the case.
```

Comment:
nitpick: This is a collective artifact, so writing commentary in first person without a note of authorship is, at best, detail without context. Let's either get your name in here, or re-write this to simply acknowledge that this might not be correct (independent of trying to make this correct, which I'll address in a separate comment).

Reply:
I reworded the comment, although I'm also curious what your thoughts were on "trying to make this correct" -- I don't see another comment about it. I was hoping for some scrutiny on this from reviewers, so thank you for that.